tests/test_validation.py

```diff
@@ -208,3 +206,120 @@ def test_validation_on_write(tmp_path):
 
     with pytest.warns(ValidationWarning):
         model.save(file_path)
+
+def test_validate_on_assignment_default():
+    model = ValidationModel()
+    assert model._validate_on_assignment==True
+
+    with pytest.warns(ValidationWarning):
+        model.meta.string_attribute = 42  # Bad assignment
+    assert model.meta.string_attribute is None
+
+
+def test_validate_on_assignment_false(tmp_path):
+    file_path = tmp_path/"test.asdf"
+
+    model = ValidationModel(validate_on_assignment=False)
+    assert model._validate_on_assignment==False
+
+    model.meta.string_attribute = 42  # Bad assignment should cause no warning
```

Comment:
In order to assert no warnings, we need to do something like this:
```python
with pytest.warns(None) as warnings:
  model.meta.string_attribute = 42
assert len(warnings) == 0
```
otherwise the test succeeds, since warnings don't interrupt the program like exceptions do.  Another option is to enable strict validation.

Reply:
I added this structure to ensure no warnings in the various tests where that is expected.